### DISCUSSION TITLE
How can I do custom authentication with Web3?

### DISCUSSION THREAD

**Discussion by trevorwrightdev** [supabase/supabase#9476] [Questions]
Hey, so right now I am trying to create Web3 authentication and integrate that with Supabase authentication in order to use RLS. Here is what I currently have working:

1. User connects their wallet 
2. User signs a message on my backend, so basically, I am able to successfully verify the user on the backend.

So here is my question:
Is there something I can send back to the frontend, from my backend, after I have verified the user on the backend, in order to use Supabase authentication? And how would the frontend work for that? I was thinking about using setAuth, but thats deprecated. Is there anything similar to that I can use?

**Comment by 0xAskar** [comment]
This may not be useful to you, but I'd recommend using [rainbowkit](https://www.rainbowkit.com/docs/introduction). That's what I've been using for my web3 apps and it's fantastic.
